### PR TITLE
chore: updates foundry-js to 0.16

### PR DESCRIPTION
**Pull request by rlucha-crowdstrike**

### CODE REVIEW

**Comment by mraible** [comment]
Duplicate of https://github.com/CrowdStrike/foundry-js-blueprint-react/pull/14.

**Comment by rlucha-crowdstrike** [comment]
@mraible I've also added the missing package-lock file here, strange that it had not been committed in the past.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,10 +7,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Use Node 20
+      - name: Use Node 23
```
Comment (mraible) [inline .github/workflows/main.yml:10]: 22 is the latest LTS. Are you sure we want to test with 23?

Comment (rlucha-crowdstrike) [reply .github/workflows/main.yml:10]: You're right, I'll use 22 then.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -7,11 +7,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Use Node 20
+      - name: Use Node 22
         uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version: 22
+          cache: 'npm'
```
Comment (mraible) [inline .github/workflows/main.yml:14]:
```suggestion
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -7,11 +7,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Use Node 20
+      - name: Use Node 22
```
Comment (mraible) [inline .github/workflows/main.yml:10]:
```suggestion
      - name: Use Node 20
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -7,11 +7,11 @@ jobs:
...
         uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version: 22
```
Comment (mraible) [inline .github/workflows/main.yml:13]:
```suggestion
          node-version: 20
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -12,6 +12,6 @@ jobs:
         with:
           node-version: 20
```
Comment (mraible) [inline .github/workflows/main.yml:13]:
```suggestion
          node-version: 20
          cache: 'npm'
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -12,6 +12,6 @@ jobs:
...
       - name: Run npm install
-        run: npm install
+        run: npm ci
```
Comment (mraible) [inline .github/workflows/main.yml:15]:
```suggestion
        run: npm install
```

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -11,6 +11,7 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: 20
+          cache: 'npm'
```
Comment (mraible) [inline .github/workflows/main.yml:14]:
```suggestion
          cache: 'npm'
          check-latest: true
```

**Pull request closed without merging**